### PR TITLE
8 ncgames

### PR DESCRIPTION
**Pull request by bakrikhalifa**

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -30,10 +31,9 @@ app.use((err, req, res, next) => {
   }
 });
 
-
 // psql error
 app.use((err, req, res, next) => {
-  if (err.code === "22P02") {
+  if (err.code === "22P02" || err.code === "23503" || err.code === "23502") {
```
Comment (k-huggs) [inline app.js]: A 23502 and 23503 aren't both 400 errors. 23503 is a 404, so you will need to have a look at how your testing has been performed.

File: controllers/controller.js [hunk 2]
```diff
@@ -48,7 +49,33 @@ exports.getCommentsByID = (req, res, next) => {
...
+      next(err);
+    });
+};
+
+exports.patchReviewByID = (req, res, next) => {
+  const updatedReviewBody = req.body;
+  const { review_id } = req.params;
+  const promises = [
+    patchReviewByIDData(review_id, updatedReviewBody),
+    checkIfCommentsExist(review_id),
```
Comment (k-huggs) [inline controllers/controller.js]: You don't need to keep adding checks to see if things exist for every end point. This is an expensive process in time and cash which should only be performed when necessary

**Review by k-huggs — COMMENTED** [review]
Have a look at the comments, same as last endpoint you don't need to check if something exists for these. The database will provide an error for you

File: app.js (hunk 1 above)
Comment (bakrikhalifa) [reply app.js]: This info was so important for me thank you. This is where my confusion was coming from regarding psql errors. Thank you again mate

**Pull request merged**